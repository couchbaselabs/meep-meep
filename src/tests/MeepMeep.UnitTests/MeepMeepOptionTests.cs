using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using MeepMeep.Testing;
using NUnit.Framework;
using MeepMeep.Extensions;

namespace MeepMeep.UnitTests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class MeepMeepOptionTests : UnitTestsOf<MeepMeepOptions>
    {
        protected override void OnTestInitialize()
        {
            SUT = new MeepMeepOptions
            {
                Nodes = new []{"http://cb1.local:8091/pools", "http://cb2.local:8091/pools"},
                Bucket = "mybucket",
                BucketPassword = "mybucketpass",
                ClusterUsername = "myclusteruser",
                ClusterPassword = "myclusterpass",
                DocKeyPrefix = "mydoc",
                DocKeySeed = 3,
                DocSamplePath = "c:\\foo\bar\\in_the_house.json",
                WorkloadSize = 42,
                NumOfClients = 13,
                MaximumConcurrencyLevel = 1,
                ThreadPoolMaxNumOfThreads = 1000,
                ThreadPoolMinNumOfThreads = 500,
                WarmupMs = 4242,
                Verbose = true
            };
        }

        [Test]
        public void When_generating_help_text_It_should_return_a_proper_help_screen()
        {
            CustomApprovals.VerifyGitFriendly(SUT.GetHelp());
        }

        [Test]
        public void When_evaluating_options_as_string_It_should_output_key_values_of_the_options()
        {
            Approvals.Verify(SUT);
        }

        [Test]
        public void Can_map_to_couchbase_config()
        {
            var clientConfig = SUT.ToClientConfig();

            clientConfig.Bucket.Should().Be(SUT.Bucket);
            clientConfig.BucketPassword.Should().Be(SUT.BucketPassword);
            clientConfig.Username.Should().Be(SUT.ClusterUsername);
            clientConfig.Password.Should().Be(SUT.ClusterPassword);
            clientConfig.Urls.Should().BeEquivalentTo(SUT.Nodes.Select(n => new Uri(n)));
        }
    }
}