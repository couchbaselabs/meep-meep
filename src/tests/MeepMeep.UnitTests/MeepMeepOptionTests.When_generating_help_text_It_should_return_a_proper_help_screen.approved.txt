MeepMeep - A super simple workload utility for the Couchbase .Net client. v0.1.0
Copyright © 2013 Daniel Wertheim

  -n, --nodes                (Default: "http://127.0.0.1:8091/pools") Space
                             separated list of nodes to connect to.

  -b, --bucket               (Default: "default") Name of the Bucket

  -p, --bucketpassword       (Default: "") Password for the bucket.

  -c, --num-clients          (Default: 1) Number of client sessions.

  -s, --wl-size              (Default: 1) The workload size. Depending on
                             Workload, it could e.g. represent the num of
                             documents to work with (per client session).

  --doc-sample-path          (Default: "") Path to a file with an UTF8 JSON
                             document, e.g used for inserts.

  --doc-key-prefix           (Default: "mm") Used as a prefix to generated
                             document keys.

  --doc-key-seed             (Default: 0) Used as a starting seed when
                             generating document keys.

  --clusteruser              (Default: "") Username for the cluster.

  --clusterpassword          (Default: "") Password for the cluster.

  --max-concurrency-level    (Default: <null>) Will control the maximum
                             concurrency level used to control how many client
                             tasks that will be executed in parallel.

  --threadpool-min           (Default: <null>) Min number of threads to be set
                             on the underlying shared threadpool.

  --threadpool-max           (Default: <null>) Max number of threads to be set
                             on the underlying shared threadpool.

  -w, --warmup-ms            (Default: 500) Determines how long time (ms) it
                             takes before a work load's operation results is
                             being seen as part of the workload.

  -v, --verbose              (Default: False) Determines detail level of output
                             generated by output writers etc.

  -h, --help                 Display this help screen.